### DISCUSSION TITLE
Fortls and compiler_commands.json

### DISCUSSION THREAD

**Discussion by endingly** [fortran-lang/fortls#289] [Ideas]
Can fortls use the compiler_commands.json generated by cmake?

**Comment by gnikit** [comment]
Not really, `fortls` knows nothing about compilers.
How would you want to use it? which settings from `compiler_commands.json` you want to propagate to `fortls`?

**Comment by endingly** [comment]
I need to extract information about other packages from the compiled database, such as `fortran-stdlib`, but the packages added via cmake fortls don't have smart completion, so I'm wondering if there's a way to get package information like clangd.

**Reply by gnikit** [reply]
You can just use regex and include the directories like `build/**` for fortls to parse whatever projects you have there. Same thing applies with `fpm`.
